File: python/cash.py
# This program will get an amount of change owed from the user and return the minimum number of coins needed
from cs50 import get_float


# Define main program function
def main():
    # Get amount of change owed from user
    change_owed = get_positive_float("Amount of change owed: ")
    # Convert to cents
    cents = int(change_owed * 100)
    # Calculate miminum number of coins
    coins = get_coins(cents)
    # Print the minimum number of coins
    print(coins)


# Define the function for getting a non negative float
def get_positive_float(prompt):
    while True:
        n = get_float(prompt)
        if n >= 0:
            return n


# Define function to calculate the minimum number of coins
def get_coins(cents):
    quarters = cents // 25
    dimes = (cents % 25) // 10
    nickels = ((cents % 25) % 10) // 5
    pennies = ((cents % 25) % 10) % 5
    return quarters + dimes + nickels + pennies


# Run main program function
main()

File: pset1/cash.c
//This program calculates the minumum number of coins needed to provide the amount of change owed.

#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // Declare variables
    float changeOwed = .41;
    int numOfCoins = 0;
    int remainder = 0;

    // Get a float input from the user as long as input is greater than 0
    do
    {
        changeOwed = get_float("Amount of change owed: ");
    }
    while (changeOwed <= 0);

    // Convert float changeOwed to integer cents
    int cents = round(changeOwed * 100);

    // Find the remainder of cents. (How many times does 25 go into cents?)
    remainder = cents % 25;

    // Calculate the number of coins (quarters) used
    numOfCoins += (cents - remainder) / 25;

    // Update the remaining amount of cents
    cents = remainder;

    // Find the remainder of cents. (How many times does 10 go into cents?)
    remainder = cents % 10;

    // Calculate the number of coins (dimes) used
    numOfCoins += (cents - remainder) / 10;

    // Update the remaining amount of cents
    cents = remainder;

    // Find the remainder of cents. (How many times does 5 go into cents?)
    remainder = cents % 5;

    // Calculate the number of coins (nickels) used
    numOfCoins += (cents - remainder) / 5;

    // Update the remaining amount of cents
    cents = remainder;

    // Find the remainder of cents. (How many times does 1 go into cents?)
    remainder = cents % 1;

    // Calculate the number of coins (pennies) used
    numOfCoins += (cents - remainder) / 1;
    
        
       // Print the total number of coins used
       printf("Number of coins used: %d\n", numOfCoins);
    
   return 0;
}